src/mixer/basetrackplayer.cpp

```diff
@@ -289,24 +291,19 @@ void BaseTrackPlayerImpl::slotTrackLoaded(TrackPointer pNewTrack,
         int reset = m_pConfig->getValueString(ConfigKey(
                 "[Controls]", "SpeedAutoReset"),
                 QString("%1").arg(RESET_PITCH)).toInt();
-        switch (reset) {
-          case RESET_PITCH_AND_SPEED:
-            // Note: speed may affect pitch
-            if (m_pRateSlider != NULL) {
-                m_pRateSlider->set(0.0);
+        if (reset == RESET_SPEED || reset == RESET_PITCH_AND_SPEED) {
+            // Avoid reseting speed if master sync is enabled and other decks with sync enabled
+            // are playing, as this would change the speed of already playing decks.
+            if (! m_pEngineMaster->getEngineSync()->otherSyncedPlaying(getGroup())) {
+                if (m_pRateSlider != NULL) {
+                    m_pRateSlider->set(0.0);
+                }
             }
-            M_FALLTHROUGH_INTENDED;
-          case RESET_PITCH:
+        }
+        if (reset == RESET_PITCH || reset == RESET_PITCH_AND_SPEED) {
```

Comment:
No, this should be a separate `if` statement. `else if` would require a third branch with duplicate code for RESET_PITCH_AND_SPEED